src/contract_client/basic_node_signer.ts

```diff
@@ -1,21 +1,33 @@
-import { Keypair, TransactionBuilder, hash } from '..'
+import { Keypair, TransactionBuilder, hash } from "..";
+import type { AssembledTransaction } from "./assembled_transaction";
+import type { ContractClient } from "./client";
 
 /**
  * For use with {@link ContractClient} and {@link AssembledTransaction}.
  * Implements `signTransaction` and `signAuthEntry` with signatures expected by
  * those classes. This is useful for testing and maybe some simple Node
  * applications. Feel free to use this as a starting point for your own
  * Wallet/TransactionSigner implementation.
+ * @param keypair {@link Keypair} to use to sign the transaction or auth entry
+ * @param networkPassphrase passphrase of network to sign for
  */
-export const basicNodeSigner = (keypair: Keypair, networkPassphrase: string) => ({
+export const basicNodeSigner = (
+  /**
+   *{@link Keypair} to use to sign the transaction or auth entry
+   */
+  keypair: Keypair,
+  /**
+   * passphrase of network to sign for
+   */
+  networkPassphrase: string,
+) => ({
+  // eslint-disable-next-line require-await
   signTransaction: async (tx: string) => {
     const t = TransactionBuilder.fromXDR(tx, networkPassphrase);
     t.sign(keypair);
     return t.toXDR();
   },
-  signAuthEntry: async (entryXdr: string): Promise<string> => {
-    return keypair
-      .sign(hash(Buffer.from(entryXdr, "base64")))
-      .toString('base64')
-  }
-})
+  // eslint-disable-next-line require-await
+  signAuthEntry: async (entryXdr: string): Promise<string> =>
```

Comment:
wait this is a good callout, if only for my own clarification - why is it marked `async`? isn't that implied by `Promise`?

Reply:
it's that whole special compile-time-vs-runtime thing! The `Promise` is _TypeScript_, and gets stripped out before any JavaScript execution (or transpilation). The `async` is JavaScript. If you have an `async` function and forget to wrap the return type in a `Promise`, TypeScript will yell at you. You gotta keep 'em in sync!